napari/util/keybindings.py

```diff
@@ -0,0 +1,582 @@
+        try:
+            _unbind_key(keybindings, seq)
+        except KeyError:
+            pass
+    elif bind_op == BindOperation.REBIND:
+        orig_seq = normalize_key_sequence(func)
+        _rebind_key(keybindings, seq, orig_seq)
+    else:
+        _bind_key(keybindings, seq, func)
+
+
+def unbind_key(keybindings, seq):
+    """Unbind a function from a keymap.
+
+    Parameters
+    ----------
+    keybindings : dict of str: callable
+        Keymap to modify.
+    seq : str
+        Key sequence.
+
+    Returns
+    -------
+    func : callable
+       Unbound function.
+
+    Raises
+    ------
+    KeyError
+        When the key sequence is not found in the keymap.
+    """
+    seq = normalize_key_sequence(seq)
+    return _unbind_key(keybindings, seq)
+
+
+def rebind_key(keybindings, new_seq, orig_seq):
+    """Rebind a function from one key sequence to another.
+
+    Parameters
+    ----------
+    keybindings : dict of str: callable
+        Keymap to modify.
+    new_seq : str
+        Key sequence to rebind to.
+    orig_seq : str
+        Key sequence to unbind from.
+
+    Raises
+    ------
+    KeyError
+        When the original key sequence is not found in the keymap.
+    """
+    new_seq = normalize_key_sequence(new_seq)
+    orig_seq = normalize_key_sequence(orig_seq)
+
+    _rebind_key(keybindings, new_seq, orig_seq)
+
+
+def _bind_keys_validate_pair(pair):
+    """Raise a TypeError if the given pair is not
+    a 2-sequence of (str, (callable, str, or None))
+    """
+    if not (isinstance(pair, Sequence) and not isinstance(pair, STRINGTYPES)):
+        raise TypeError(f'expected {pair} to be a sequence, got {type(pair)}')
+    key, op = pair
+    if not isinstance(key, str):
+        raise TypeError(f'key {key} must be a string, not {type(key)}')
+    _determine_binding_op(op)
+
+
+def unbind_keys(keybindings, *seqs, error=True):
+    """Unbind functions from a keymap.
+
+    Parameters
+    ----------
+    keybindings : dict of str: callable
+        Keymap to modify.
+    *seqs : iterable of str
+        Key sequences to unbind. If not provided, unbind all key sequences.
+    error : bool
+        Whether to error when a function is not found in the keymap.
+        Defaults to ``True``.
+
+    Returns
+    -------
+    funcs : list of callable
+        Unbound functions.
+
+    Raises
+    ------
+    KeyError
+        When a function is not found in the keymap
+        and ``error`` is set to ``True``.
+    """
+    if len(seqs) == 0:
+        seqs = list(keybindings.keys())
+    elif len(seqs) == 1 and isinstance(seqs[0], Iterable):
+        seqs = seqs[0]
+
+    funcs = []
+
+    for seq in seqs:
+        try:
+            funcs.append(unbind_key(keybindings, seq))
+        except KeyError:
+            if error:
+                # rebind unbound functions
+                bind_keys(keybindings, zip(seqs[:len(funcs)], funcs))
+                raise
+
+    return funcs
```

Comment:
This is a bit of a crazy API... I'm wondering if `unbind_keys` should not coexist with `unbind_key` — just have one function that deals with both use cases. I'd prefer if we didn't need to return `funcs` either... I can see where it might be useful, but we should try to keep our API minimal to begin with, and grow it where needed, rather than start with an enormous API.